beacon-chain/types/crystallized_state.go

```diff
@@ -289,11 +289,11 @@ func (c *CrystallizedState) NewStateRecalculations(aState *ActiveState, block *B
 
 		justifiedSlot, finalizedSlot, justifiedStreak = casper.FinalizeAndJustifySlots(slot, justifiedSlot, finalizedSlot,
 			justifiedStreak, blockVoteBalance, c.TotalDeposits())
+	}
 
-		newState.data.Crosslinks, err = newState.processCrosslinks(aState.PendingAttestations(), slot, newState.Validators(), block.SlotNumber())
-		if err != nil {
-			return nil, err
-		}
+	newState.data.Crosslinks, err = newState.processCrosslinks(aState.PendingAttestations(), newState.Validators(), block.SlotNumber())
```

Comment:
I discussed this in discord a week or so ago. `processCrosslinks` should be called once per state recalculation, not per slot.